Walking mushroom expansion

Tokiko1 opened the pull request:
There's a lot of changes to walking mushrooms in this! I'll try to list them all!

*Abilities added! Walking mushrooms can now learn a few abilities that they can use in combat.
There's a stun attack adds a stun chance, a freezing bite that cools you down(feed them icepeppers), a teleporting ability which allows them to teleport to their enemy(feed them bluespace tomatoes), a environment smash upgrade(feed them soybeans), a regeneration + speed upgrade and a spaceproofing ability that stops them from dying and allows them to move freely in space.

*Walking mushrooms can now evolve into a new form. They currently turn into a white, capless version of the original sprite. If anyone can sprite a more menacing looking walking mushroom, please do so I can replace the current sprite. 
![evolvedmushroom1](https://cloud.githubusercontent.com/assets/7812993/3944272/0d597594-25f7-11e4-9c36-d97b2938e398.gif)
![evolved2](https://cloud.githubusercontent.com/assets/7812993/3944273/0e8735aa-25f7-11e4-87b5-bc7281f9f50a.gif)

*Adds walking mushroom seeds to hacked seedvendors to encourage the crew to hold mushroom fights as these events can be quite entertaining.

*Leveling rewritting. You can now keep leveling your mushroom even if it is higher level than the losing one, but the chance to level up decreases the bigger the level difference becomes. Also a few bug fixes and a lot of stat rebalances, mainly for the next feature.

*You can feed ghost chillies to your walking mushroom to make them attack the crew instead! Any mushroom fed with those will attack everyone except the person that fed them. Rampaging mushrooms will still attack other mushrooms but will no longer eat them, making further leveling impossible! They also explode in a rather harmless but scary fireblast when they die so it's impossible to revive them with lazarus injectors/mushrooms or anything else. Mushrooms turned hostile are logged for admins!

*You can now use a plant analyzer on your mushroom to have it display its stats!

![mushroom_data_scan](https://cloud.githubusercontent.com/assets/7812993/3944269/bb1a8e1c-25f6-11e4-990b-3d8913c25de3.PNG)

*Also fixes the misnamed dead spacewasp sprite.

*Added sugarcane seeds to the vending machine. I have no idea how long they have been missing.

Tokiko1 committed:
*Greatly expands walking mushrooms by adding new abilities, rewritten stats and leveling and the ability to set them loose on the station to fight the crew.

*Fixes missing dead wasp sprite.

Tokiko1 committed:
Rebalance and fix.

Lumipharon commented:
Oh yes, this is great. Need to be able to name your mushrooms though, if you can't already.

Tokiko1 committed:
Fixes.

Tokiko1 committed:
*Adds walking mushroom seeds to hacked seed vendors to encourage the crew to have some more mushroom fights.

*Adds the long missing sugarcane seeds to the seed vendor aswell!

*Fixes some smaller bugs and prevents lazarus injector weirdness.

Tokiko1 commented:
Alright this is merge ready!

ike709 commented:
@RemieRichards ^
